sharks/src/share_ff.rs

```diff
@@ -4,11 +4,15 @@ use core::convert::TryInto;
 #[cfg(feature = "fuzzing")]
 use arbitrary::Arbitrary;
 
+#[cfg(feature = "zeroize_memory")]
+use zeroize::Zeroize;
+
 use crate::ff::*;
 
 pub const FIELD_ELEMENT_LEN: usize = 32;
 
 #[cfg_attr(feature = "fuzzing", derive(Arbitrary))]
+#[cfg_attr(feature = "zeroize_memory", derive(Zeroize))]
```

Comment:
I'm not very familiar with the zeroize crate. Doesn't `derive(Zeroize)` just implement the `zeroize()` method on the object, which the user needs to explicitly call? I don't see any new calls in the add-rs crate. Would `derive(ZeroizeOnDrop)` be better here?

Reply:
Yes, having looked into that a little bit more, I think you're correct. Unfortunately, I don't think we can do that because it conflicts with the base field implementation which derives `Copy`. However, I think that, actually, the `Share` structs don't actually contain any secret data. Rather, they are simply random values that are only meaningful when you combine `K` of them together. Otherwise, they are distributed identically to random values.

So, in summary, using `zeroize` in this context may not even be required. What do you think @gpestana? The alternative would be to do as @rillian says and add some explicit `.zeroize()` calls in the adss-rs crate.

Reply:
They suggest implementing `ZeroizeOnDrop` [directly on a wrapper type](https://github.com/RustCrypto/traits/blob/e5b9920/elliptic-curve/src/ecdh.rs#L167-L170), to handle the member types as we see fit. that is, solve it by adding another level of indirection.

Maybe that's reasonable to the extent that the field element is just a number. 32 bytes is on the edge of what I'd consider 'small' for these purposes, but it is bounded and often the optimiser does a fine job eliminating redundant copies even with moderate-sized structs. In our case, explicit `.zeroize()` calls might be simpler.

Curious what @gpestana thinks.

Reply:
Bummer that `ZeroizeOnDrop` cannot be derived here. I think that wrapping `Fp` on a new struct and directly handling `.zeroize()` would be a good option here, although it seems to require quite a few code changes.

OTOH, as Alex mentioned, the `Shares` are meaningless out of context and do not hold secret data (as far as `K` is not known), so I will leave you to decide which direction to take. I'm good with either options.

Reply:
Where would we implement the `.zeroize()` calls in this library? My impression is that these only really make sense when calling the `sta-rs` code in a high-level application? For example, in `star-wasm` we could call `.zeroize()` after generating the shares, but otherwise I don't see any place where we would explicitly call. Is this the correct way to think about it? Or should we be calling `.zeroize()` in the internal code as well.

In either case, maybe just implementing the `ZeroizeOnDrop` wrapper is the best thing here so that we don't have to leave this up to implementers?

Reply:
If we're not treating `Share` as sensitive, perhaps there's nothing to do in this particular library. At best the caller might want benefit from a `zeroize()` method implementation as you had before so they don't have to wrap it.

Rather, I'd look at where we might have *k* shares collected together. if they're passed by reference or returned by value, that's fine, but the idea of ZeroizeOnDrop is to not leave any intermediate copies laying around in de-allocated memory. It's possible `interpolate` is doing that, but hopefully not. I'd have to step through it to be sure.

More clear is  `Sharks::recover()` which copies the `x` values from the shares into a hash set and clones a matching set of `Share` structs into a local `Vec` for interpolation. When that's dropped, a complete set of data for recovering the key is left sitting on the heap. If we're serious about implementing Zeroize internally, those two variables would be a good candidates.